### PR TITLE
update to get show_scatterer_slices to work

### PR DESCRIPTION
**Pull request by anna-wang**
show_scatterer_slices wasn't working. update includes:
- changing voxelated scatterer from ndarray to xarray
- removing the rescaling by vmin and vmax for voxelated scatterers. This rescaling causes the visualisation to show incorrectly because the same vmin and vmax are used again in imshow.

### CODE REVIEW

File: holopy/scattering/scatterer/scatterer.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def index_at(self, points, background=0):
         index = np.ones_like(domains, dtype=dtype) * background
         for i, n in enumerate(ns):
             index[domains==i+1] = n
-        return index
+        return xr.DataArray(index, attrs = {'_image_scaling':[0,self.n.max()], 'voxel':1})
```
Comment (barkls) [inline holopy/scattering/scatterer/scatterer.py:102]: The `self.n.max()` call is tripping up a few unittests on travis. Specifically, it looks like n is sometimes defined as a float, list, or tuple in tests so the max call is failing. Wrapping `self.n` in `hp.core.metadata.ensure_array` should solve.

Comment (barkls) [inline holopy/scattering/scatterer/scatterer.py:102]: This is a bit of a missed opportunity to take full advantage of xarray by keeping track of the coordinates of each voxel. It should be pretty straightforward to pass in coords and dims according to the [xarray documentation](http://xarray.pydata.org/en/stable/data-structures.html#creating-a-dataarray). Let me know if you want help implementing this, but I think we should be able get everything working and merged in without it if you prefer.

File: holopy/core/io/vis.py [hunk 2]
```diff
@@ -341,7 +341,7 @@ def show_scatterer_slices(scatterer, spacing):
         voxel spacing for the visualization
     """
     vol = scatterer.voxelate(spacing, 0)
-    show2d(vol)
+    Show2D(vol)
```
Comment (barkls) [inline holopy/core/io/vis.py:344]: You should try calling `hp.core.io.vis.display_image` before passing to `Show2D` That function is meant to massage any type of input to a standardized xarray output for visualization or saving. Hopefully that removes the need to check if im has a `voxel` attribute later on. `display_image` is a bit messy, so feel free to ask me if you aren't sure of anything.

**Comment by barkls** [comment]
Hey @anna-wang any progress on this? If you are satisfied just having something that works and would rather pass on extending it any further, that's completely understandable and someone else can work on elaborating on your fix (probably me or @briandleahy). In that case, it would be very helpful if you can provide a minimal example of what was broken before. If you would like to keep working on this, there's no rush, and I'm happy to help with any questions or difficulties that come up!

**Comment by barkls** [comment]
It's possible some overzealous protections were preventing you from pushing to this branch, even though you created it. I've just relaxed those protections.

**Comment by anna-wang** [comment]
@barkls - I'm so sorry it's taken me so long to look into this! I really appreciate your comments though, turns out my messy workaround was totally unnecessary and calling hp.core.io.vis.display_image was all that was needed! Thanks so much for looking into it.

**Comment by barkls** [comment]
I'm glad it ended up being a simple fix and I'm happy that `hp.core.io.vis.display_image` is doing its job! It would still be nice to get a test that captures the failing behaviour. Let me know if you want to write one or if we should assign it to someone else.

**Comment by anna-wang** [comment]
@barkls sorry again for the delay! I know it's simple but I'm a bit swamped, if you could assign to someone else that would be great. This is the code that was failing previously.

```
import holopy as hp
from holopy.scattering.scatterer import Indicators, Sphere, Scatterer
import numpy as np

s1 = Sphere(r = .5, center = (0, 0, 0))
s2 = Sphere(r = .51, center = (0, 0, 0))

def cap(point):
    return(np.logical_and(np.logical_and(point[...,2] > 0, s2.contains(point)),
           np.logical_not(s1.contains(point))))
indicators = Indicators([s1.contains, cap],
                        [[-.51, .51], [-.51, .51], [-.51, .51]])
janus = Scatterer(indicators, (1.34, 2.0), (5, 5, 5))

hp.core.io.vis.show_scatterer_slices(janus,spacing=0.005)
```

I believe this also failed:
```
import holopy as hp
from holopy.scattering.scatterer import Indicators, Sphere, Scatterer
import numpy as np

s = Sphere(r = .5, center = (0, 0, 0))

hp.core.io.vis.show_scatterer_slices(s,spacing=0.005)
```

Thank you!

**Comment by barkls** [comment]
Great, thanks for the examples! I'll look into it...

**Comment by barkls** [comment]
closing in favour of #347

**Pull request closed without merging**